### PR TITLE
Fix `delete_experiment` and `restore_experiment` in FileStore implementation

### PR DESCRIPTION
**Pull request by mariusschlegel**
## Related Issues/PRs

 #8177

## What changes are proposed in this pull request?

(Please fill in changes proposed in this fix)

## How is this patch tested?

- [x] Existing unit/integration tests
- [ ] New unit/integration tests
- [ ] Manual tests (describe details, including test results, below)

## Does this PR change the documentation?

- [x] No. You can skip the rest of this section.
- [ ] Yes. Make sure the changed pages / sections render correctly in the documentation preview.

### Is this a user-facing change?

- [ ] No. You can skip the rest of this section.
- [x] Yes. Give a description of this change to be included in the release notes for MLflow users.

(Details in 1-2 sentences. You can just refer to another PR with a description if this PR is part of a larger change.)

Fix persistent updating of `lifecycle_stage` of experiments and related runs when deleting/restoring experiments with FileStore backend (#8178, @mariusschlegel).

### What component(s), interfaces, languages, and integrations does this PR affect?

Components

- [ ] `area/artifacts`: Artifact stores and artifact logging
- [ ] `area/build`: Build and test infrastructure for MLflow
- [ ] `area/docs`: MLflow documentation pages
- [ ] `area/examples`: Example code
- [ ] `area/model-registry`: Model Registry service, APIs, and the fluent client calls for Model Registry
- [ ] `area/models`: MLmodel format, model serialization/deserialization, flavors
- [ ] `area/recipes`: Recipes, Recipe APIs, Recipe configs, Recipe Templates
- [ ] `area/projects`: MLproject format, project running backends
- [ ] `area/scoring`: MLflow Model server, model deployment tools, Spark UDFs
- [ ] `area/server-infra`: MLflow Tracking server backend
- [x] `area/tracking`: Tracking Service, tracking client APIs, autologging

Interface

- [ ] `area/uiux`: Front-end, user experience, plotting, JavaScript, JavaScript dev server
- [ ] `area/docker`: Docker use across MLflow's components, such as MLflow Projects and MLflow Models
- [ ] `area/sqlalchemy`: Use of SQLAlchemy in the Tracking Service or Model Registry
- [ ] `area/windows`: Windows support

Language

- [ ] `language/r`: R APIs and clients
- [ ] `language/java`: Java APIs and clients
- [ ] `language/new`: Proposals for new client languages

Integrations

- [ ] `integrations/azure`: Azure and Azure ML integrations
- [ ] `integrations/sagemaker`: SageMaker integrations
- [ ] `integrations/databricks`: Databricks integrations

<a name="release-note-category"></a>

### How should the PR be classified in the release notes? Choose one:

- [ ] `rn/breaking-change` - The PR will be mentioned in the "Breaking Changes" section
- [ ] `rn/none` - No description will be included. The PR will be mentioned only by the PR number in the "Small Bugfixes and Documentation Updates" section
- [ ] `rn/feature` - A new user-facing feature worth mentioning in the release notes
- [x] `rn/bug-fix` - A user-facing bug fix worth mentioning in the release notes
- [ ] `rn/documentation` - A user-facing documentation change worth mentioning in the release notes

### CODE REVIEW

File: mlflow/store/tracking/file_store.py [hunk 1]
```diff
@@ -429,7 +429,19 @@ def delete_experiment(self, experiment_id):
...
+        deletion_time = get_current_time_millis()
+        experiment._set_last_update_time(deletion_time)
+        runs = self._list_run_infos(experiment, experiment_id)
+        for run in runs:
+            run_id = run.info.run_id
+            run_info = self._get_run_info(run_id)
+            if run_info is None:
+                raise MlflowException(
+                    "Run '%s' metadata is in invalid state." % run_id, databricks_pb2.INVALID_STATE
+                )
```
Comment (WeichenXu123) [inline mlflow/store/tracking/file_store.py:442]: I think we don't need to raise error here but printing warning instead.

File: mlflow/store/tracking/file_store.py [hunk 2]
```diff
@@ -463,7 +475,18 @@ def restore_experiment(self, experiment_id):
...
+            new_info = run_info._copy_with_overrides(lifecycle_stage=LifecycleStage.ACTIVE)
+            self._overwrite_run_info(new_info, deleted_time=None)
```
Comment (WeichenXu123) [inline mlflow/store/tracking/file_store.py:489]: Could you add test for fixed case in delete_experiment and restore_experiment ?

File: mlflow/store/tracking/file_store.py (hunk 1 above)
Comment (mariusschlegel) [reply mlflow/store/tracking/file_store.py]: > I think we don't need to raise error here but printing warning instead.

Done.

File: mlflow/store/tracking/file_store.py (hunk 2 above)
Comment (mariusschlegel) [reply mlflow/store/tracking/file_store.py]: > Could you add test for fixed case in delete_experiment and restore_experiment ?

Done.

**Review by WeichenXu123 — APPROVED** [review]
LGTM

File: mlflow/store/tracking/file_store.py [hunk 3]
```diff
@@ -1,86 +1,86 @@
...
 from mlflow.utils.time_utils import get_current_time_millis
-from mlflow.utils.uri import (
-    append_to_uri_path,
-    resolve_uri_if_local,
+from mlflow.utils.uri import append_to_uri_path, resolve_uri_if_local
+from mlflow.utils.validation import (
+    _validate_batch_log_data,
+    _validate_batch_log_limits,
+    _validate_experiment_id,
+    _validate_experiment_name,
```
[the remark this answers is not in the document]
Comment (mariusschlegel) [reply mlflow/store/tracking/file_store.py]: OK, I have reverted the sorting of the import statements.

[the remark this answers is not in the document]
Comment (WeichenXu123) [reply mlflow/store/tracking/file_store.py]: Thank you!

**Pull request merged**